pkg/loop/internal/broker.go

```diff
@@ -141,8 +141,10 @@ func (b *brokerExt) serve(name string, server *grpc.Server, deps ...resource) (u
 
 func (b *brokerExt) closeAll(deps ...resource) {
 	for _, d := range deps {
-		if err := d.Close(); err != nil {
-			b.Logger.Error(fmt.Sprintf("Error closing %s", d.name), "err", err)
+		if d.Closer != nil {
```

Comment:
handle case when the resource is declared but not uninitialized

Reply:
🤔 This doesn't feel like it should be necessary, and I wonder if this will hide incorrect instantiations in the future. How come you had to add this?

Reply:
Ah, I guess it's due to this: https://github.com/smartcontractkit/chainlink-relay/pull/166/files#diff-28ea9d6d17fb7499fbbd0f83828a8335ac28eacff629256d75b261bdb2ee123aR147?